### DISCUSSION TITLE
population.plot_composition()

### DISCUSSION THREAD

**Discussion by jolinmaj** [skfolio/skfolio#25] [Ideas]
Is there a way to pull data from the population object to create plots of portfolio weights thorugh time

**Comment by HugoDelatte** [comment]
The best way to represent a portfolio with changing weights is with `MultiPeriodPortfolio`.
You can directly call `plot_composition()` from a `MultiPeriodPortfolio`. 
Also, if you create a  `Population` containing  `MultiPeriodPortfolio` and call `population.plot_composition()` , each multi period portfolio weights will be displayed like in [this example](https://skfolio.org/auto_examples/5_clustering/plot_5_nco_grid_search.html#prediction)

If you need some custum display, you can access each portfolio of the population [like this](https://github.com/skfolio/skfolio/blob/1b312ac8180434a2df1d1e415b7f1be75ce43b02/src/skfolio/population/_population.py#L467)

Let me know if that helps.

**Comment by jolinmaj** [comment]
I am not sure this will work without creating a portfolio at each date and then obtaining the weights?

If I am using InverseVolatility, for example, as the volatility changes , so the weights would need to be updated. And I suppose this feeds into how often I re-balance. I don't see this in the MuliPeriodPortfolio?
